examples/typehints/inventory.py

```diff
@@ -0,0 +1,25 @@
+from atom.api import Atom
+from atom.api import observe
+
+
+class InventoryItem(Atom):
+    """Class for keeping track of an item in inventory."""
+
+    name: str
+    unit_price: float
+    quantity_on_hand: int = 0
+
+    def total_cost(self) -> float:
+        return self.unit_price * self.quantity_on_hand
+
+    @observe("unit_price")
+    def _observe_unit_price(self, change):
```

Comment:
You are requesting to register the observer twice (special name and decorator) also you need a type hint to appease mypy